api/String.cpp

```diff
@@ -652,9 +652,9 @@ void String::replace(const String& find, const String& replace)
 		}
 	} else if (diff < 0) {
 		unsigned int size = len; // compute size needed for result
+		diff = 0 - diff;
```

Comment:
Ah, this line hurts 😉 . (Could it not be `abs(diff)`? However, you did not commit this crime, simply moved the line to the right position 👍 .